### ISSUE TITLE
Add option to choose the loss function in `compute_loss` function. 

### ISSUE THREAD

**Issue by abhi-glitchhg** [pytorch/vision#5983]
### 🚀 The feature

In the [compute loss](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/fcos.py#L54) function,  `giou_loss`  is used to calculate the loss by default [see here](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/fcos.py#L96), but now as `CIOU` and `DIOU` loss functions are added to torchvision, maybe we could let the user choose in between the loss functions

cc @datumbox @YosuaMichael, @NicolasHug,

### Motivation, pitch

This gives little more freedom to the user to choose between the loss functions.

**Comment by datumbox** [comment]
@abhi-glitchhg It seems you've uncovered my evil plan... 😄 

The reason why I've asked you to add ciou/diou was to test them on RetinaNet. This functionality was already added on the specific model (see #5756) but it's private because we don't want to commit to the API yet. I've kicked off a PR to experiment with it at #5984 and see if it brings improvements. It doesn't do exactly what you want cause it's not on FCOS but it sets the basis.
